src/Microsoft.Sbom.Api/Config/ConfigSanitizer.cs

```diff
@@ -106,14 +114,28 @@ private ConfigurationSetting<AlgorithmName> GetHashAlgorithmName(IConfiguration
             };
         }
 
-        private ConfigurationSetting<string> GetNamespaceBaseUriFromAssembly(IConfiguration configuration, ILogger logger)
+        private ConfigurationSetting<string> GetNamespaceBaseUri(IConfiguration configuration, ILogger logger)
         {
-            // If assembly name is not defined returned the current value.
-            if (string.IsNullOrWhiteSpace(assemblyConfig.DefaultSBOMNamespaceBaseUri))
+            // If assembly name is not defined but a namespace was provided, then return the current value.
+            if (string.IsNullOrWhiteSpace(assemblyConfig.DefaultSBOMNamespaceBaseUri) && !string.IsNullOrEmpty(configuration.NamespaceUriBase?.Value))
             {
                 return configuration.NamespaceUriBase;
             }
 
+            // If assembly name is not defined and namespace was not provided then return the default namespace as per spdx spec https://spdx.github.io/spdx-spec/v2.2.2/document-creation-information/#653-examples.
+            if (string.IsNullOrWhiteSpace(assemblyConfig.DefaultSBOMNamespaceBaseUri) && string.IsNullOrEmpty(configuration.NamespaceUriBase?.Value))
+            {
+                string defaultNamespaceUriBase = "https://spdx.org/spdxdocs/sbom-tool-" + SBOMToolVersion + "-" + Guid.NewGuid();
+
+                logger.Information($"No namespace URI base provided, using default value {defaultNamespaceUriBase}");
+
+                return new ConfigurationSetting<string>
```

Comment:
This may be out of scope for this particular PR, but it seems like a constructor that accepts a Value and a Source (and defaults the value of the Source to `SettingSource.Default` might be a good idea.  As it is, need to "remember" to set the source type seems like a bug waiting to happen.

Reply:
Confirmed that if no Source is provided then telemetry picks it up as "Source=Default".